### PR TITLE
DOCSUP-791: Docs for optimize_read_in_order setting

### PR DESCRIPTION
**Pull request by olgarev**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category
- Documentation for #8130

### CODE REVIEW

**Comment by BayoNet** [comment]
We have a topic describing the `ORDER BY` clause - https://clickhouse.tech/docs/en/sql-reference/statements/select/order-by/

I think it's better to move the *Usage* section of the setting description there under the *Optimizations* section.

File: docs/en/sql-reference/statements/select/order-by.md [hunk 1]
```diff
@@ -70,6 +70,22 @@ Running a query may use more memory than `max_bytes_before_external_sort`. For t
 
 External sorting works much less effectively than sorting in RAM.
 
+## Optimize_read_in_order Setting {#optimize_read_in_order}
```
Comment (BayoNet) [inline docs/en/sql-reference/statements/select/order-by.md:73]:
```suggestion
## Optimization of Data Reading {#optimize_read_in_order}
```

File: docs/en/sql-reference/statements/select/order-by.md [hunk 2]
```diff
@@ -70,6 +70,22 @@ Running a query may use more memory than `max_bytes_before_external_sort`. For t
...
+
+ If `ORDER BY` expression has coinciding prefix with the table sorting key, you can optimize the query by using the [optimize_read_in_order](../../../operations/settings/settings.md#optimize_read_in_order) setting.
```
Comment (BayoNet) [inline docs/en/sql-reference/statements/select/order-by.md:75]:
```suggestion
 If `ORDER BY` expression has a prefix that coincides with the table sorting key, you can optimize the query by using the [optimize_read_in_order](../../../operations/settings/settings.md#optimize_read_in_order) setting.  
```

File: docs/en/sql-reference/statements/select/order-by.md [hunk 3]
```diff
@@ -70,6 +70,22 @@ Running a query may use more memory than `max_bytes_before_external_sort`. For t
...
+ 
+ When the `optimize_read_in_order` setting is enabled, the Clickhouse server uses the table index and reads the data in order of the `ORDER BY` key. This allows to avoid reading all data in case of specified [LIMIT](../../../sql-reference/statements/select/limit.md). So queries on big data with small limit are processed faster.
+
+Optimization works with both `ASC` and `DESC` and doesn't work together with [GROUP BY](../../../sql-reference/statements/select/group-by.md) clause and [FINAL](../../../sql-reference/statements/select/from.md#select-from-final) modifier.
+
+When the `optimize_read_in_order` setting is disabled, the Clickhouse server does not use the table index while processing `SELECT` queries.
+
+There are queries that have `ORDER BY` clause, large `LIMIT` and [WHERE](../../../sql-reference/statements/select/where.md) condition that requires to read huge amount of records before queried data is found. For these queries you should consider disabling `optimize_read_in_order` manually.
```
Comment (BayoNet) [inline docs/en/sql-reference/statements/select/order-by.md:83]:
```suggestion
Consider disabling `optimize_read_in_order` manually, when running queries that have `ORDER BY` clause, large `LIMIT` and [WHERE](../../../sql-reference/statements/select/where.md) condition that requires to read huge amount of records before queried data is found.
```

File: docs/en/sql-reference/statements/select/order-by.md [hunk 4]
```diff
@@ -70,6 +70,22 @@ Running a query may use more memory than `max_bytes_before_external_sort`. For t
...
+
+Optimization is supported in [MergeTree](../../../engines/table-engines/mergetree-family/mergetree.md) engine, and also in [Merge](../../../engines/table-engines/special/merge.md), [Buffer](../../../engines/table-engines/special/buffer.md) and [MaterializedView](../../../engines/table-engines/special/materializedview.md) storages with underlying `MergeTree` tables.
```
Comment (BayoNet) [inline docs/en/sql-reference/statements/select/order-by.md:85]:
```suggestion
Optimization is supported in the following table engines:

- [MergeTree](../../../engines/table-engines/mergetree-family/mergetree.md)
- [Merge](../../../engines/table-engines/special/merge.md)
- [Buffer](../../../engines/table-engines/special/buffer.md) 
- [MaterializedView](../../../engines/table-engines/special/materializedview.md) storages with underlying `MergeTree` tables
```

File: docs/en/sql-reference/statements/select/order-by.md [hunk 5]
```diff
@@ -70,6 +70,22 @@ Running a query may use more memory than `max_bytes_before_external_sort`. For t
...
+
+For `MaterializedView` storage the optimization works with saved queries like `SELECT ... FROM merge_tree_table ORDER BY pk`. But it is not supported in the queries like `SELECT ... FROM view ORDER BY pk` if the saved query doesn't have `ORDER BY` clause.
```
Comment (BayoNet) [inline docs/en/sql-reference/statements/select/order-by.md:87]:
```suggestion
In `MaterializedView`-egine tables the optimization works with saved queries like `SELECT ... FROM merge_tree_table ORDER BY pk`. But it is not supported in the queries like `SELECT ... FROM view ORDER BY pk` if the saved query doesn't have `ORDER BY` clause.
```

File: docs/en/sql-reference/statements/select/order-by.md [hunk 6]
```diff
@@ -70,6 +70,27 @@ Running a query may use more memory than `max_bytes_before_external_sort`. For t
...
+Consider disabling `optimize_read_in_order` manually, when running queries that have `ORDER BY` clause, large `LIMIT` and [WHERE](../../../sql-reference/statements/select/where.md) condition that requires to read huge amount of records before queried data is found.
+
+Optimization is supported in the following table engines:
+
+- [MergeTree](../../../engines/table-engines/mergetree-family/mergetree.md)
+- [Merge](../../../engines/table-engines/special/merge.md)
+- [Buffer](../../../engines/table-engines/special/buffer.md) 
+- [MaterializedView](../../../engines/table-engines/special/materializedview.md) storages with underlying `MergeTree` tables
+
+In `MaterializedView`-egine tables the optimization works with saved queries like `SELECT ... FROM merge_tree_table ORDER BY pk`. But it is not supported in the queries like `SELECT ... FROM view ORDER BY pk` if the saved query doesn't have `ORDER BY` clause.
```
Comment (blinkov) [inline docs/en/sql-reference/statements/select/order-by.md:92]:
```suggestion
In `MaterializedView`-engine tables the optimization works with views like `SELECT ... FROM merge_tree_table ORDER BY pk`. But it is not supported in the queries like `SELECT ... FROM view ORDER BY pk` if the view query doesn't have the `ORDER BY` clause.
```

**Pull request merged**